### ISSUE TITLE
Create developer install TestPlan.md document

### ISSUE THREAD

**Issue by johnsaigle** [aces/Loris#4172]
We should create a document defining steps to take to verify the install of a  _developer_ instance of LORIS. This document would be followed by a developer during our release testing to determine that it is easy and straight-forward to install a dev instance. 

This could include steps to take during post-install, such as 
* "Bootstrapping": installing additional dependencies for the dev tools
* Getting a working copy of test data (i.e. the "Raisinbread" data set)
* ...

---

Some other things: 
* This document would need to be thoroughly vetted each release to be up-to-date with our dependencies, etc. This would help us toward our goal of reducing the need for ad-hoc Wiki hotfixes, which are a necessary evil while there are gaps in our documentation.

* The document should be as general as possible and should not contain information pertaining to our internal IT procedures.

* There should also be a companion document for Vagrant installs, if specific information is needed for that context.

* There is existing information available to help with this task: the "LORIS 101" document, existing Wiki pages, as well as institutional knowledge not written down but known by members of the team. There are likely some common bugs as well that can be fixed as we codify this process.

**Comment by johnsaigle** [comment]
#4168 is relevant to this ticket. `phan` has a dependency on the PHP `ast` extension. This is required only for developer instances of LORIS and so is not installed by default. It should be done automatically on dev installs or at the very least explicitly documented for developer installs.

**Comment by johnsaigle** [comment]
https://github.com/aces/Loris/issues/3313 is related. Need to make sure Vagrant install works effortlessly.

**Comment by johnsaigle** [comment]
@ridz1208 Following up on the discussion in #4168 would you be able to do a small talk to the group about the status of the Wiki? What branch it's on, tentative schedule, how to contribute, etc?

**Comment by johnsaigle** [comment]
#4365 is an attempt at automating some of the post-install steps and satisfying (Ubuntu) system dependencies as LORIS moves forward.

**Comment by johnsaigle** [comment]
@kchatpar and I discovered more issues for Ubuntu-based developer installs today.

* Fresh Ubuntu lack the following tools by default:
    * make
    * npm

* The version of `node` shipped with Debian (and therefore Ubuntu) is very out of date to the point where it does not work with webpack. To remedy this we updated node using the following commands
```
curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
sudo apt-get install -y nodejs
```

These steps will likely be important for the bootstrap script, the Vagrant file, as well as the Wiki.

**Comment by johnsaigle** [comment]
CentOS configuration is probably related to this https://github.com/aces/Loris/issues/3946

**Comment by johnsaigle** [comment]
Note that these issues just came up again with Clark, our new volunteer.

**Comment by christinerogers** [comment]
> * The version of `node` shipped with Debian (and therefore Ubuntu) is very out of date to the point where it does not work with webpack. To remedy this we updated node using the following commands

John's March 12 comment appears to be an issue for users installing the 21 release, not documented in the Release notes or Readme. Adding it to Doc project. 
@johnsaigle do you feel this should be split into a new issue?

**Comment by johnsaigle** [comment]
Yes I think that point specifically is probably worthy of being treated as its own bugfix givent hat the release is already out.

**Comment by christinerogers** [comment]
Thanks for #5017 👍

**Comment by johnsaigle** [comment]
I feel confident for the most part that this is covered by what I've written in #5253.

**Comment by johnsaigle** [comment]
I've changed #5253 to focus on production. A new document will be needed for modifications needed for a development installation.

**Comment by christinerogers** [comment]
This issue is being worked on presently by our GSOC2020 team (led by) @AlexandraLivadas , @h-karim and @spell00 also with @laemtl <br>
This issue will be resolved by a combination of the following PRs: <br>
a. for the Loris 23 release: 
* #6510 - install markdowns updated for dev and prod
* #6511 - installation Troubleshooting guide

b. later in summer 2020: 
* future PR to flesh out `99_developer_guide/` markdowns (in the "new wiki"), based on GSOC2020 training materials - to provide useful info for LORIS developers out in the wild.

**Comment by laemtl** [comment]
@christinerogers Can we benefit from @haoweiqiu help on this?

**Comment by haoweiqiu** [comment]
Hi @christinerogers and @laemtl,
I think I could help with this. Could you provide me with more details on how I may help? Thanks!

**Comment by laemtl** [comment]
@driusan @christinerogers Do we still support Vagrant?

**Comment by driusan** [comment]
No

**Comment by laemtl** [comment]
@haoweiqiu This document is a good starting point, and needs modifications for a development installation.

**Comment by ridz1208** [comment]
@driusan @christinerogers and others who might care about this

My instinct is to close this issue, here is why:

1. We have [official install documentation](https://github.com/aces/Loris/tree/main/docs/wiki/00_SERVER_INSTALL_AND_CONFIGURATION/01_LORIS_Install/Ubuntu) so we don't need a new one, if changes are required they should be made there
2. Every new developer ends up making their own documentation which ends up being 95% reusable with other developers but requiring 5% environemnt differences that makes the next new hire rewrite another new documentation.... the cycle keeps repeating
3. the description asks for documentation which is not specific to MCIN but includes several aspecs that are specific to MCIN. Raisinbread for example is a public dataset but installed VERY differently for MCIN developers vs non-mcin developers and I would suspect non-MCIN developers to be a rather rare usecase no warranting a whole document written for them? (they wouldnt even use RB since RB is mainly used internally)
4. Same logic applies to multiple other MCIN vs non-MCIN developer differences, this is why the developer install documentation always resided in the google drive and not on the branch (since it wouldnt work for non-MCIN developers with access to the repo)


If we are talking about a google doc... this issue does not belong here and MULTIPLE google docs already exist so we dont need the issue
